python_modules/dagster/dagster/_core/definitions/partition_mapping.py

```diff
@@ -217,24 +107,51 @@ class IdentityPartitionMapping(PartitionMapping, NamedTuple("_IdentityPartitionM
     partitions in the downstream asset to the same partition in the upstream asset.
     """
 
-    def get_upstream_partitions_for_partition_range(
+    def get_upstream_mapped_partitions_result_for_partitions(
         self,
-        downstream_partition_key_range: Optional[PartitionKeyRange],
-        downstream_partitions_def: Optional[PartitionsDefinition],
+        downstream_partitions_subset: Optional[PartitionsSubset],
         upstream_partitions_def: PartitionsDefinition,
-    ) -> PartitionKeyRange:
-        if downstream_partitions_def is None or downstream_partition_key_range is None:
+        current_time: Optional[datetime] = None,
+        dynamic_partitions_store: Optional[DynamicPartitionsStore] = None,
+    ) -> UpstreamPartitionsResult:
+        if downstream_partitions_subset is None:
             check.failed("downstream asset is not partitioned")
 
-        return downstream_partition_key_range
+        if downstream_partitions_subset.partitions_def == upstream_partitions_def:
```

Comment:
Previously the `IdentityPartitionMapping` would raise an error when partition keys in `downstream_partitions_subset` did not exist in `upstream_partitions_def`. This change switches this behavior to return the list of nonexistent but required parent keys instead.

A good future change would be to add more definition-time checking, i.e. to ensure that when the IdentityPartitionMapping is applied or inferred, the partitions defs are the same.